tool2schema/schema.py

```diff
@@ -75,6 +75,104 @@ def SaveGPTEnabled(module: ModuleType, path: str, schema_type: SchemaType = Sche
     json.dump(schemas, open(path, "w"))
 
 
+class ParseException(Exception):
+    """Exception for schema parsing errors."""
+
+    pass
+
+
+def LoadGPTEnabled(
+    module: ModuleType,
+    function: dict[str, str],
+    validate: bool = True,
+    ignore_hallucinations: bool = True,
+) -> Optional[tuple[Callable, dict]]:
+    """
+    Given a function dictionary containing the name of a function and the arguments to pass to it
+    in the form of a JSON string, retrieve the corresponding function among those with the
+    `GPTEnabled` decorator defined in `module`. When `validate` is true, validate the arguments
+    string and raise `ParseException` if the arguments are not valid and cannot be used to invoke
+    the function.
+
+    :param module: The module where the function is defined
+    :param function: A dictionary with keys `name` and `arguments`, where `name` is the name of
+        the function to find, and `arguments` is a JSON string representing the values to pass to
+        the function as arguments
+    :param validate: Whether to validate the function arguments
+    :param ignore_hallucinations: When true, any hallucinated arguments are ignored; when false,
+        an exception is raised if any hallucinated arguments are found
+    :return: A tuple consisting of the function and a dictionary of argument values, or None if no
+        function with the expected name and the `GPTEnabled` decorator is found
+    :raises ParseException: If the argument string is not valid, meaning it is not parsable as JSON,
+        it can be parsed but is not a dictionary, it does not include all required arguments, it
+        includes values that are not of the expected type, or it contains hallucinated arguments
+        and `ignore_hallucinations` is false
+    """
+
+    if not (name := function.get("name", None)):
+        raise ParseException("Function name is missing from the dictionary")
+
+    if not (arguments := function.get("arguments", None)):
+        raise ParseException("Function arguments are missing from the dictionary")
+
+    try:
+        arguments = json.loads(arguments)
+
+    except json.decoder.JSONDecodeError:
+        raise ParseException("Arguments are not in valid JSON format")
+
+    if type(arguments) is not dict:
+        raise ParseException("Arguments are not in the form of a dictionary")
+
+    f = FindGPTEnabledByName(module, name)
+
+    if not f:
+        # A function with the given name was not found
+        return None
```

Comment:
```suggestion
        return None, None
```

Otherwise, code will crash if you want to do something like

```python
f, args = LoadGPTEnable(...)
```

Reply:
I had it raise an exception with [900ede5](https://github.com/cadifyai/tool2schema/pull/59/commits/900ede5c0a9a6ef698e240478c88fb267b66402a)